advocacy_docs/partner_docs/HashicorpVault/05-UsingHashicorpVault.mdx

```diff
@@ -0,0 +1,241 @@
+
+!!! Note
+    It is important to note that this doc is intended for versions 15.2 of EDB Postgres Advanced Server as this version has the necessary addition of TDE.
+
+Before you can implement Hashicorp Vault Secrets Engine with your EDB Postgres Advanced Server 15.2 instance, you must ensure that you have the following downloaded to your system:
+- Python
+- [pykmip](https://pypi.org/project/PyKMIP/#files)
+- edb_tde_kmip_client.py downloaded from your EDB Repos access
+
+All of the `.pem` files that you created need to be on the system where your EDB Postgres Advanced Server 15.2 instance is installed.  For this guide, all of the `.pem` files are in the `/tmp/` directory as well as the `edb_tde_kmip_client.py` file.
+
+## Create pykmip.conf File
+
+1. On your system where you have your EDB Postgres Advanced Server 15.2 instance, navigate to the directory where you have saved your *.pem files and the edb_tde_kmip_client.py client.
+
+2. In that directory create a file called `pykmip.conf` and input the following:
+- Host
+- Port
+- Username
+- Password
+- Keyfile
+- Certfile
+- Ca_certs
+
+In the below example, Username and Password are commented out as Hashicorp Vault does not utilize those for their connectivity:
+```bash
+ubuntu@ip-172-31-46-134:/tmp$ cat pykmip.conf 
+[client]
+host=127.0.0.1
+port=5696
+#username=     
+#password=
+keyfile=/tmp/key.pem
+certfile=/tmp/cert.pem
+ca_certs=/tmp/ca.pem
+#cert_reqs=CERT_REQUIRED
+```
+
+!!! Note
+    For more information on the pykmip.conf file and the contents of it you can visit the [pykmip documentation](https://pykmip.readthedocs.io/en/latest/client.html).
+
+## Create a Key on Hashicorp Vault Secrets Engine
+
+1. On your system where you have your EDB Postgres Advanced Server 15.2 instance, assume root user to create the key on the Hashicorp Vault Secrets Engine.
+
+2. Type `python3` and then input the following, making adjustments per your system setup and directory paths:
+
+```bash
+>>> from kmip.pie import client
+>>> from kmip import enums
+>>> c = client.ProxyKmipClient(config_file='/tmp/pykmip.conf')
+>>> c.open()
+>>> key_id = c.create(enums.CryptographicAlgorithm.AES, 128, name='edbtestkey')
+>>> c.activate(key_id)
+>>> key_id
+>>> 'key_output_shows_here'
+>>> c.close()
+
+```
+3. If this runs without error then your key has been successfully created.  You cannot view keys that you create in Hashicorp Vault.
+
+## Verify Encryption and Decryption
+
+To ensure that your key that you created will be able to encrypt and decrypt data, run the following two commands as the root user on your system where you have your EDB Postgres Advanced Server 15.2 instance.
+
+1. `printf secret | python3 /tmp/edb_tde_kmip_client.py encrypt --out-file=test.bin --pykmip-config-file=/tmp/pykmip.conf  --key-uid='key_output_here’ --variant=pykmip`
```

Comment:
Should tell them what these are

Location of  KMP Client : /tmp/edb_tde_kmip_client.py
Output file  : test.bin
Location of pykmip configureation file : /tmp/pkykmip.con
etc.